runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -0,0 +1,343 @@
+            }
+
+            return next != null;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public Entry<K, V> next() {
+            if (hasNext()) {
+                current = next;
+                next = null;
+                return current;
+            } else {
+                throw new NoSuchElementException();
+            }
+
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int size() {
+        return dataSetSize.get();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean isEmpty() {
+        return dataSetSize.get() == 0;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean containsKey(@NonNull Object key) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+        try {
+            byte[] value = rocksDb.get(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes());
+            return value != null;
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * Please use {@link StreamingMap#entryStream()}.
+     */
+    @Override
+    public boolean containsValue(@NonNull Object value) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V get(@NonNull Object key) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+
+        try {
+            byte[] value = rocksDb.get(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes());
+            if (value == null) {
+                return null;
+            }
+            return (V) serializer.deserialize(Unpooled.wrappedBuffer(value), corfuRuntime);
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public V put(@NonNull K key, @NonNull V value) {
+        final ByteBuf keyPayload = Unpooled.buffer();
+        final ByteBuf valuePayload = Unpooled.buffer();
+        serializer.serialize(key, keyPayload);
+        serializer.serialize(value, valuePayload);
+
+        try {
+            rocksDb.put(
+                    keyPayload.array(), keyPayload.arrayOffset(), keyPayload.readableBytes(),
+                    valuePayload.array(), valuePayload.arrayOffset(), valuePayload.readableBytes());
+        } catch (RocksDBException ex) {
+            throw new UnrecoverableCorfuError(ex);
+        } finally {
+            keyPayload.release();
+            valuePayload.release();
+        }
+
+        dataSetSize.incrementAndGet();
```

Comment:
LSM cannot compute the size of elements until full merge happens. So do we know if this value is accurate? We might have to do a rocksDb.get() to see if the value exists and then selectively increment right? Please correct me if this is already handled. thanks

Reply:
https://github.com/facebook/rocksdb/releases
you can get an estimate but count isn't accurate in RocksDB...
https://stackoverflow.com/questions/25538750/getting-total-number-of-key-value-pairs-in-rocksdb

Reply:
For now I am using `RocksDB::keyMayExist(...)` which is more light weight. Technically, in order to be compliant with the `Map` interface we should be returning the value to which this map previously associated the key,